arch/x86_64/dynamic.S

```diff
@@ -106,7 +107,7 @@ GLOBAL(__dentry__)
 
 	add $48, %rsp
 	.cfi_adjust_cfa_offset -48
-
+	popf
```

Comment:
Do we really need to save and restore the flags?  Then we should do it for normal mcount as well, no?

Reply:
It should work without it if we assume that the flag register is always set in the function before using it. If it's set by the caller, mcount_entry may change its state and undefined behavior may occur when the flag is used in the callee.
IMO we are better safe than sorry.

Reply:
Yeah, but then it's not a problem of dynamic tracing.  The normal mcount call should consider that also IMO.  But I've never seen the compiler generating that code for mcount (caller) and mcount itself (in [glibc](https://code.woboq.org/userspace/glibc/gmon/mcount.c.html)).  So I assumed the flag is not propagated to a different function.

With that being said, I agree with your safety concern.  But please seperate it from this change and cover the other cases (mcount, fentry) as well.  Also I think it'd be easier to handle if you save it after the normal registers so that you can leave the offsets unchanged.